src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -8,9 +10,25 @@ namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.NamingStyles
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase
     {
-        public CSharpNamingStyleDiagnosticAnalyzer()
+        protected sealed override void OnCompilationStartAction(
+            CompilationStartAnalysisContext context,
+            ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>> idToCachedResult)
         {
+            // HACK: RegisterSymbolAction doesn't work with local functions
+            context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.LocalFunctionStatement);
 
+            void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
```

Comment:
can you have an explicit "return" before this local function, then add a ``// local functions``` comment.  it makes it much easier to tell that htese are locals, and not methods.

Reply:
fixed